code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/components/SmallImageCard.kt

```diff
@@ -50,16 +50,20 @@ fun SmallImageCard(
     style: SmallImageUIStyle,
     observer: AepUIEventObserver?,
 ) {
+    var isLoading by remember { mutableStateOf(true) }
     var imageBitmap by remember { mutableStateOf<Bitmap?>(null) }
     val imageUrl = if (isSystemInDarkTheme() && ui.getTemplate().image?.darkUrl != null)
         ui.getTemplate().image?.darkUrl else ui.getTemplate().image?.url
 
     LaunchedEffect(ui.getTemplate().id) {
         observer?.onEvent(UIEvent.Display(ui))
-        if (imageUrl != null) {
+        if (imageUrl.isNullOrBlank()) {
+            isLoading = false
+        } else {
             imageBitmap = withContext(Dispatchers.IO) {
                 UIUtils.downloadImage(imageUrl)
             }
+            isLoading = false
```

Comment:
should this be isLoading = true as the image is being downloaded in the else branch?

Reply:
`isLoading` has default value of true. Once the coroutine to download the image completes, this line will run and set isLoading to false

Reply:
my bad, i had missed the default value above